src/main/scala/PropertyChecker.scala

```diff
@@ -406,7 +399,20 @@ object PropertyChecker {
             }
           }
         } else if (PropertyCheckerConstants.DateFormatDataTypes.contains(baseValue)) {
-          throw new NotImplementedError() // Implement after adding DateFormat class
+          if (objectNode.get("format").isTextual) {
+            try {
+              val format = DateFormat(Some(objectNode.get("format").asText()), None).getFormat()
```

Comment:
Feels a bit messy here, a lot going on in the same line. Split it out for we have the `Some(...)` stuff in a variable?

Reply:
Ya agreed, I have changed it